src/Cmfd.cpp

```diff
@@ -3123,6 +3123,15 @@ void Cmfd::initialize() {
       omp_init_lock(&_cell_locks[r]);
     omp_init_lock(&_edge_corner_lock);
 
+    /* Compute and log size in memory of CMFD matrices */
+    int num_rows = _num_cmfd_groups * _local_num_xn * _local_num_yn *
+                   _local_num_zn;
+    int num_non_zero_coeffs = 7 * 10; // 10 is est. number of scatter groups
```

Comment:
Should be 7 + 10 here, since scattering only happens within a cell. That means the flux of neighbor cells won't contribute to the scattering source of the current cell. And the peak memory is 7+_num_cmfd_groups during the matrix construction because the full scattering matrix is assumed.
Actually, I have an idea to use CSR or CSC to store the sparse scattering matrix for memory reduction.

Reply:
You're right.
CSR is already used for the CMFD A matrix. 
I think that would work, but the savings might be limited. In an assembly, for CMFD materials 230*289*70*70*4bytes / 1e6 is only 1.3GB

Reply:
I made some analysis to memory usage in current CMFD, and promoted some possible idea to reduce it. Below is the usage for a typical Beavrs assembly with 289*230*70
![image](https://user-images.githubusercontent.com/34581354/48927047-dfb29180-eea0-11e8-8a08-20d3370e0b33.png)